indico/modules/rb_new/client/js/common/bookings/OccurrenceActionsDropdown.jsx

```diff
@@ -0,0 +1,180 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import ReactDom from 'react-dom';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {Form as FinalForm, Field} from 'react-final-form';
+import {Button, Icon, Confirm, Dropdown, Form, Portal, TextArea} from 'semantic-ui-react';
+
+import {ReduxFormField, formatters, validators as v} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+import * as bookingsActions from './actions';
+
+import './OccurrenceActionsDropdown.module.scss';
+
+
+class OccurrenceActionsDropdown extends React.Component {
+    static propTypes = {
+        booking: PropTypes.object.isRequired,
+        date: PropTypes.string.isRequired,
+        actions: PropTypes.exact({
+            changeBookingOccurrenceState: PropTypes.func.isRequired,
+            fetchBookingDetails: PropTypes.func.isRequired
+        }).isRequired,
+    };
+
+    state = {
+        activeConfirmation: null,
+        dropdownOpen: false,
+        top: 0,
+        left: 0,
+    };
+
+    hideConfirm = () => {
+        this.setState({activeConfirmation: null});
+    };
+
+    showConfirm = (type) => {
+        this.setState({activeConfirmation: type});
+    };
+
+    changeOccurrenceState = (action, data = {}) => {
+        const {date} = this.props;
+        const {booking: {id}, actions: {changeBookingOccurrenceState, fetchBookingDetails}} = this.props;
+        changeBookingOccurrenceState(id, date, action, data).then(() => {
+            fetchBookingDetails(id);
+        });
+    };
+
+    findPositioning = () => {
+        const positioning = ReactDom.findDOMNode(this).getBoundingClientRect();
+        return {
+            top: positioning.top,
+            left: positioning.left
+        };
+    };
+
+    onClickButton = () => {
```

Comment:
I think the more common naming scheme for react event handlers is `handleButtonClick` (not that we have much consistency there anyway ;))